src/item_use.c

```diff
@@ -1284,7 +1284,7 @@ void ItemUseInBattle_BagMenu(u8 taskId)
     else
     {
         PlaySE(SE_SELECT);
-        if (!(B_TRY_CATCH_TRAINER_BALL >= GEN_4 && (ItemId_GetBattleUsage(gSpecialVar_ItemId) == EFFECT_ITEM_THROW_BALL) && (gBattleTypeFlags & BATTLE_TYPE_TRAINER)))
+        if (!(B_TRY_CATCH_TRAINER_BALL >= GEN_4 && (ItemId_GetBattleUsage(gSpecialVar_ItemId) == EFFECT_ITEM_THROW_BALL) && (gBattleTypeFlags & BATTLE_TYPE_TRAINER)) && ItemId_GetBattleUsage(gSpecialVar_ItemId) != EFFECT_ITEM_USE_POKE_FLUTE)
```

Comment:
I would check the importance or the bag pocket instead of hardcoding the effect. This also means that regular items with this effect would not be expendable.